dialer_test.go

```diff
@@ -69,6 +69,15 @@ func TestDialerInstantiationErrors(t *testing.T) {
 	if err == nil {
 		t.Fatalf("expected NewDialer to return error, but got none.")
 	}
+	_, err = NewDialer(context.Background(),
```

Comment:
These changes could probably be in a separate PR?

Reply:
I'll send them in a separate PR.